src/prif.f90

```diff
@@ -34,6 +38,9 @@ module prif
   public :: prif_atomic_define_int, prif_atomic_define_int_indirect, prif_atomic_define_logical, prif_atomic_define_logical_indirect
   public :: prif_atomic_ref_int, prif_atomic_ref_int_indirect, prif_atomic_ref_logical, prif_atomic_ref_logical_indirect
 
+  type, public :: prif_event_type
+  end type
```

Comment:
These four empty type declarations make me uneasy, because it's not clear to a casual reader that they are looking at a non-conforming/unimplemented feature, or what eventually belongs here.

How about for now we fill each unimplemented type declaration with a placeholder such as the following:
```suggestion
  type, public :: prif_event_type
    private
    ! TODO: actual implementation
    integer :: unimplemented_feature_placeholder = 0
  end type
```
This should convey to human readers that the feature is unimplemented, and to compilers that the components are private and default-initialized as required by the PRIF spec.